workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/flows/redeem/ConfidentialRedeemFungibleTokensFlow.kt

```diff
@@ -19,25 +19,31 @@ import net.corda.core.transactions.SignedTransaction
  * @param issuerSession session with the issuer tokens should be redeemed with
  * @param observerSessions optional sessions with the observer nodes, to witch the transaction will be broadcasted
  * @param additionalQueryCriteria additional criteria for token selection
+ * @param changeHolder optional change key, if using accounts you should generate the change key prior to calling this
+ *                     flow then pass it in to the flow via this parameter
  */
 class ConfidentialRedeemFungibleTokensFlow
 @JvmOverloads
 constructor(
         val amount: Amount<TokenType>,
         val issuerSession: FlowSession,
         val observerSessions: List<FlowSession> = emptyList(),
-        val additionalQueryCriteria: QueryCriteria? = null
+        val additionalQueryCriteria: QueryCriteria? = null,
+        val changeHolder: AbstractParty? = null
 ) : FlowLogic<SignedTransaction>() {
     @Suspendable
     override fun call(): SignedTransaction {
-        // Send anonymous identity to the issuer.
-        issuerSession.send(TransactionRole.PARTICIPANT)
-        observerSessions.forEach { it.send(TransactionRole.OBSERVER) }
-        val changeOwner = subFlow(ProvideKeyFlow(issuerSession))
+        // If a change holder key is not specified then one will be created for you. NB. If you want to use accounts
+        // with tokens, then you must generate and allocate the key to an account up-front and pass the key in as the
+        // "changeHolder".
+        val confidentialHolder = changeHolder ?: let {
```

Comment:
Do you send it to the issuer?

Reply:
Good question. It gets synced here: https://github.com/corda/token-sdk/blob/781b93212ecee550db035a3a7130cc207171f3fc/workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/flows/redeem/AbstractRedeemTokensFlow.kt#L55